compass/Coordinator.java

```diff
@@ -353,37 +361,90 @@ private void createAndBroadcastMilestone(String trunk, String branch) throws Arg
    */
   private boolean validateTransactionsToApprove(String trunk, String branch) {
     if (validatorAPIs.size() > 0) {
-      boolean isConsistent = validatorAPIs.parallelStream().allMatch(api -> {
+
+      return validatorAPIs.parallelStream().allMatch(validatorApi -> {
         CheckConsistencyResponse response;
         try {
-          response = api.checkConsistency(trunk, branch);
+          response = getCheckConsistencyResponseWithRetires(trunk, branch, validatorApi);
           if (!response.getState()) {
-            log.error("{} reported invalid consistency: {}", api.getHost(), response.getInfo());
+            log.error("{} reported invalid consistency: {}", validatorApi.getHost(), response.getInfo());
           }
           return response.getState();
-        } catch (ArgumentException e) {
+        } catch (InterruptedException e) {
           e.printStackTrace();
           return false;
         }
       });
-
-      return isConsistent;
     }
 
     //nothing was checked so validation can't fail
     return true;
   }
 
+  private GetNodeInfoResponse getNodeInfoWithRetries() throws InterruptedException {
+    GetNodeInfoResponse response = null;
+    for(int i = 0; i < config.APIRetries; i++) {
+      try {
+        response = api.getNodeInfo();
+        break;
+      } catch (IllegalStateException | ArgumentException e) {
+        e.printStackTrace();
+        Thread.sleep(config.APIRetryInterval);
+      }
+    }
+    if (response == null) {
+      throw new RuntimeException("getNodeInfo failed, check node!");
+    }
+
+    return response;
+  }
+
+  private GetTransactionsToApproveResponse getGetTransactionsToApproveResponseWithRetries() throws InterruptedException {
+    GetTransactionsToApproveResponse response = null;
+    for(int i = 0; i < config.APIRetries; i++) {
+      try {
+        response = api.getTransactionsToApprove(depth, state.latestMilestoneHash);
```

Comment:
We should deal with what Jota actually does when a 500 / timeout happens etc. Having a wrapper function for these retries will also help us dealing with these occurrences.